shared-bindings/gamepad/GamePad.c

```diff
@@ -108,6 +109,7 @@ STATIC mp_obj_t gamepad_make_new(const mp_obj_type_t *type, size_t n_args,
     if (!gamepad_singleton) {
         gamepad_singleton = m_new_obj(gamepad_obj_t);
         gamepad_singleton->base.type = &gamepad_type;
+        gamepad_singleton = gc_make_long_lived(gamepad_singleton);
```

Comment:
I don't quite get what you are saying here. Sorry, can you be more explicit?

Reply:
The pointer is not being saved to the root pointer location. You need to use `MP_STATE_VM(gamepad_singleton) =` instead of `gamepad_singleton =`.

You won't need the declaration above so I suggest removing it first. After its removed, the compile will be broken and adding the `MP_STATE_VM` portion will fix it.

Reply:
But then I need to use `MP_STATE_VM(gamepad_singleton)` everywhere?

Reply:
You can have local pointers to make code more concise but you should set it once here and read it elsewhere.

Reply:
This could be:

```C
if (!MP_STATE_VM(gamepad_singleton)) {
    gamepad_obj_t* gamepad_singleton = m_new_obj(gamepad_obj_t);
    gamepad_singleton->base.type = &gamepad_type;
    gamepad_singleton = gc_make_long_lived(gamepad_singleton);
    MP_STATE_VM(gamepad_singleton) = gamepad_singleton;
}
```

Reply:
I see, thanks.